### ISSUE TITLE
Classify func unused

### ISSUE THREAD

**Issue by codeninja55** [ShokuninSan/som#1]
Hi,

Thanks for making this SOM implementation. Just wondering why you never used the classify function to do the Euclidean distance? Or have I missed something.

I am fairly new to Scala so have not noticed whether you actually used it.

**Comment by ZeroHypotheses** [comment]
Hi @codeninja55,

I'm glad that you're interested in using the SOM. Seems that I missed to write a test for that so that people can actually see how to use it. I just wrote a short hint on the README. Sorry for that.

However, `classify` is a method a user might call on a fitted SOM instance to identify a best matching unit (BMU) to an unseen data point. There is no point (at least I could think of) in using this method within the library itself. 

Hope that helps. 

Just let me know if you have any further questions. Please note that it could take a couple of days for you to get an answer.

**Comment by codeninja55** [comment]
Can you give an example of how I would use it?

**Comment by ZeroHypotheses** [comment]
Be aware that a SOM is not primarily used for classification. 

More usual applications are projections of high-dimensional input to lower dimensions (i.e. two), clustering or visualization. 

However, you can use `SOM.classify` to assign a new data point to it's closest neighbor in that low dimensional space. That way you can infer to which region/cluster - on the two dimensional surface - your new data point belongs to.

Try to run the RGB example provided in this repository, add a line calling `som.classify` with new RGB values as documented in the README and see if the results make sense to you. 

You can also find plenty of use-cases for dimensionality reduction, clustering, etc. on the web. For example, download the Iris flower data set create a train/test split, fit the SOM on the training set and classify on the test set. I'd love to see a pull-request for that ;)
